src/ci/run.sh

```diff
@@ -45,6 +45,8 @@ fi
 ci_dir=`cd $(dirname $0) && pwd`
 source "$ci_dir/shared.sh"
 
+export CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
```

Comment:
Hmm, why does this need to be set both here and in the `ci.yml` file? I would expect it to only need to be set here.

Reply:
The environment isn't inherited inside Docker images. I could use the `--env` flag to forward it, but I felt less confident that would cover everything.